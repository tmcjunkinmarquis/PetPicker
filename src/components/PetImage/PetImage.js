import React from 'react';
import './PetImage.css';
import PropTypes from 'prop-types';

const PetImage = (props) => {

  const petUrl = props.pet.pic;
  return (
    <div >
      <img className="responsive-img"
        src={petUrl}
        alt=''
      ></img>
    </div>
  );
};

export default PetImage;

PetImage.propTypes = {
  name: PropTypes.string,
<<<<<<< HEAD
  pic: PropTypes.string,
  display: PropTypes.string,
  pet: PropTypes.object
};
=======
  pic: PropTypes.string
};
>>>>>>> welcome page, petpicking pages, styled
