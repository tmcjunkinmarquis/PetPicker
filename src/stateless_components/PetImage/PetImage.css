.pet-image {
  width: 300px;
  height: 350px;
  border: 2px solid blue;
}

.hidden {
  display: none;
}