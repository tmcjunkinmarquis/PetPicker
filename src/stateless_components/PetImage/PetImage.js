import React, { Component } from 'react';
import './PetImage.css'

const PetImage = (props) => {


  return (
    

    <img ></img>


  )
}

export default PetImage;