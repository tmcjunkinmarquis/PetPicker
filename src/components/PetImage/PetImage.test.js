import React from 'react';
import { shallow } from 'enzyme';
import PetImage from './PetImage';

describe('PetImage', () => {
  let wrapper;
  let mockPet;
  let mockPetUrl;

  beforeEach(() => {
    mockPet = { id: 1, name: "Fido", pic: "https://www.google.com/url?sa=i&rct=j&q=&esrc=s&source=images&cd=&cad=rja&uact=8&ved=2ahUKEwi1ytO44LrdAhUE44MKHRt_B4YQjRx6BAgBEAU&url=https%3A%2F%2Fpixabay.com%2Fen%2Fphotos%2Fdog%2F&psig=AOvVaw2MhxK_sIy2XjmduZrqIH17&ust=1537023384535700" }

    mockPetUrl = mockPet.pic

    wrapper = shallow(<PetImage pet={mockPet} />)
  });



  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});