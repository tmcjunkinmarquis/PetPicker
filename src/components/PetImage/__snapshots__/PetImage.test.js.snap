// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetImage should match a snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PetImage
    pet={
        Object {
            "id": 1,
            "name": "Fido",
            "pic": "https://www.google.com/url?sa=i&rct=j&q=&esrc=s&source=images&cd=&cad=rja&uact=8&ved=2ahUKEwi1ytO44LrdAhUE44MKHRt_B4YQjRx6BAgBEAU&url=https%3A%2F%2Fpixabay.com%2Fen%2Fphotos%2Fdog%2F&psig=AOvVaw2MhxK_sIy2XjmduZrqIH17&ust=1537023384535700",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt=""
        src="https://www.google.com/url?sa=i&rct=j&q=&esrc=s&source=images&cd=&cad=rja&uact=8&ved=2ahUKEwi1ytO44LrdAhUE44MKHRt_B4YQjRx6BAgBEAU&url=https%3A%2F%2Fpixabay.com%2Fen%2Fphotos%2Fdog%2F&psig=AOvVaw2MhxK_sIy2XjmduZrqIH17&ust=1537023384535700"
/>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "",
        "src": "https://www.google.com/url?sa=i&rct=j&q=&esrc=s&source=images&cd=&cad=rja&uact=8&ved=2ahUKEwi1ytO44LrdAhUE44MKHRt_B4YQjRx6BAgBEAU&url=https%3A%2F%2Fpixabay.com%2Fen%2Fphotos%2Fdog%2F&psig=AOvVaw2MhxK_sIy2XjmduZrqIH17&ust=1537023384535700",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt=""
          src="https://www.google.com/url?sa=i&rct=j&q=&esrc=s&source=images&cd=&cad=rja&uact=8&ved=2ahUKEwi1ytO44LrdAhUE44MKHRt_B4YQjRx6BAgBEAU&url=https%3A%2F%2Fpixabay.com%2Fen%2Fphotos%2Fdog%2F&psig=AOvVaw2MhxK_sIy2XjmduZrqIH17&ust=1537023384535700"
/>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "https://www.google.com/url?sa=i&rct=j&q=&esrc=s&source=images&cd=&cad=rja&uact=8&ved=2ahUKEwi1ytO44LrdAhUE44MKHRt_B4YQjRx6BAgBEAU&url=https%3A%2F%2Fpixabay.com%2Fen%2Fphotos%2Fdog%2F&psig=AOvVaw2MhxK_sIy2XjmduZrqIH17&ust=1537023384535700",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
